src/trainer.c

```diff
@@ -14,78 +14,92 @@
 #include "stage.h"
 
 
-void trainer_init(Trainer *tnr) {
-	memset(tnr, 0, sizeof(*tnr));
-	// store two sets of values
-	// one is "per stage"
-	// the other is for the entire game session
-	tnr->trainer_lives_stage = 0;
-	tnr->trainer_bombs_stage = 0;
-	tnr->trainer_hits_stage = 0;
-
-	tnr->trainer_lives_total = 0;
-	tnr->trainer_bombs_total = 0;
-	tnr->trainer_hits_total = 0;
+void trainer_init(Trainer *trainer) {
+	// sets the default values
+	memset(trainer, 0, sizeof(*trainer));
 }
 
-
 bool trainer_enabled(void) {
-	if (config_get_int(CONFIG_TRAINER_MODE) && config_get_int(CONFIG_TRAINER_STATS)) {
-		if (config_get_int(CONFIG_TRAINER_LIVES) || config_get_int(CONFIG_TRAINER_BOMBS) || config_get_int(CONFIG_TRAINER_INVULN)) {
-			return 1;
-		}
+	if (config_get_int(CONFIG_TRAINER_MODE)) {
+		return true;
+	}
+	return false;
```

Comment:
Actually, why not just return the expression inside the `if` from those functions? Same with the new `player_can_bomb`, etc.

Another style nitpick that I missed: we don't put spaces before the opening parenthesis after `if`/`for`/`while`, sorry.